subsys/net/ip/net_context.c

```diff
@@ -1260,15 +1287,34 @@ static int context_sendto(struct net_context *context,
 		return -EBADF;
 	}
 
-	if (!dst_addr &&
+	if (sendto && addrlen == 0 && dst_addr == NULL && buf != NULL) {
+		/* User wants to call sendmsg */
+		msghdr = buf;
+	}
+
+	if (!msghdr && !dst_addr &&
 	    !(IS_ENABLED(CONFIG_NET_SOCKETS_CAN) &&
 	      net_context_get_ip_proto(context) == CAN_RAW)) {
 		return -EDESTADDRREQ;
 	}
 
 	if (IS_ENABLED(CONFIG_NET_IPV6) &&
 	    net_context_get_family(context) == AF_INET6) {
-		struct sockaddr_in6 *addr6 = (struct sockaddr_in6 *)dst_addr;
+		const struct sockaddr_in6 *addr6 =
```

Comment:
could you make that change in a separate patch? same for addr4 below. (both in the same patch)

Reply:
Hmm, not sure why we would need to separate this one. Now this is also bisectable.

Reply:
because that has nothing really directly related to current patch but instead is a fix for an existing issue

Reply:
Ah, you meant the const stuff, sure I can do that separately. I though you referred to other changes around these lines.

Reply:
> Ah, you meant the const stuff, sure I can do that separately.

+1, thanks @tbursztyka for suggesting that.

Reply:
I separated the const thingy and after analysing the change, I am not sure this was really needed after all as we were not really fixing anything here.